app/bundles/PageBundle/Controller/PublicController.php

```diff
@@ -147,7 +148,7 @@ public function indexAction(
                         if (!empty($variantCookie)) {
                             if (isset($variants[$variantCookie])) {
                                 // if not the parent, show the specific variant already displayed to the visitor
-                                if ($variantCookie !== $entity->getId()) {
+                                if ((int) $variantCookie !== $entity->getId()) {
```

Comment:
Why do we need this casting?

Reply:
Yes, because the return type of the `$request->cookies->get()` is `bool|float|int|string|InputBag`. Want to make sure we are safe, when the cookie has a numeric string and strict equals will work as we expect. 
Best would be to have a check before this if with something like `if (!is_int($variantCookie) || !is_string($variantCookie) || !is_numeric($variantCookie)) { throw \RuntimeException }`. WDYT?

Reply:
Closing this conversation, because `$variantCookie` here can be `string|int` only, because array only accepts those AFAIK. And `$entity->getId()` will produce the `int` type.

Reply:
I've seen many times that `$entity->getId()` returns `string`. And it should have if the ID is type of `BIGINT unsigned` as that can return a number greater than max PHP `int` number.

Reply:
@escopecz Yes, sure there are cases where this line does not make sense, but in this case we have `\Mautic\PageBundle\Entity\Page::getId`, which returns an int (according to PHPDoc), also the `\Mautic\CoreBundle\Doctrine\Mapping\ClassMetadataBuilder::addId` adds an unsigned int and not a bigint.
Also, i'm not against changing this to `(string) $variantCookie !== (string) $entity->getId()`

Reply:
Do not trust PHPDocs. They were copied from one place to another without any thought if they are true or not. I think that retyping both to the same type would make sense. I'm not sure if it matters if it is to int or string in this case.

Reply:
Made it a string cast, to be very safe.